exercises/saddle-points/introduction.md

```diff
@@ -7,5 +7,5 @@ You need to analyze each grid on the map to find good trees for your tree house.
 
 A good tree is both:
 
-- taller than every tree to the east and west, so that you have the best possible view of the sunrises and sunsets.
-- shorter than every tree to the north and south, to minimize the amount of tree climbing.
+- no shorter than any tree to the east and west, so that you have the best possible view of the sunrises and sunsets.
+- no taller than any tree to the north and south, to minimize the amount of tree climbing.
```

Comment:
I'm not sure this is an improvement to be honest. Negatives are harder to read than positives in general.

Reply:
I agree. Sadly though there is no single word synonym of «no shorter». The previous «larger» has confused people by being false.

Reply:
Was that because it could also be of the same length?

Reply:
Yes.

«At least as tall as all» might be the next most succinct.

Reply:
I like that better!

Reply:
Does it need to be succinct?

"The tree must be equal in height to the tallest tree to the north and south. No other trees can be taller to the north and south. This minimizes climbing."

Reply:
It just needs to be clear, so I'm happy to have a less succinct description.

> The tree must be equal in height to the tallest tree to the north and south

To me this reads as if it couldn't be taller.

Reply:
«the tallest tree» also suggests (again!) that there is only one.

Reply:
What about:

"The tree must either be:
- taller than the other trees to the north and south, or
- as tall as the tallest tree to the north and south.
 
No other trees can be taller to the north and south. This minimizes climbing."

Not sure about the formatting, but I'd like the fact that we're explicit about the two variants.

Reply:
The «at least as tall as all» variant suggested earlier:

```suggestion
- at least as tall as all trees to the east and west, so that you have the best possible view of the sunrises and sunsets.
- at most as tall as all tree to the north and south, to minimize the amount of tree climbing.
```

The more verbose variant suggested most recently:

```suggestion
- either
  - taller than the other trees to the east and west, or
  - as tall as the tallest trees to the east and west.

  No other trees can be taller to the east and west.
  This produces the best possible view of the sunrises and sunsets.
- either
  - shorter than the other trees to the north and south, or
  - as tall as the tallest trees to the north and south.

  No other trees can be shorter to the north and south.
  This minimizes climbing.
```

A note about this last one: it tripped up both Erik (the original is wrong) and me (I was insecure correcting and re-correcting it). The above is only a naive rendering; there might well be a better one.

---

For the moment I prefer one of «no shorter than any» and «at least as tall as all». Comparison:

- Pro «no shorter than any»: people like to think in terms of violated rules. This formulation focuses on how a tree might violate the requirements: by being too short.
- Contra «no shorter than any»: it contains a negation.
- Pro «at least as tall as all»: it does not contain a negation.
- Contra «at least as tall as all»: less immediately clear how this requirement could be violated.